modules/terraform-zscc-gwlb-aws/main.tf

```diff
@@ -16,6 +17,18 @@ resource "aws_lb_target_group" "gwlb_target_group" {
     healthy_threshold   = var.healthy_threshold
     unhealthy_threshold = var.unhealthy_threshold
   }
+
+  target_failover {
+    on_deregistration = var.rebalance_enabled == true ? "rebalance" : "no_rebalance"
+    on_unhealthy      = var.rebalance_enabled == true ? "rebalance" : "no_rebalance"
+  }
+
+  # type attribute only applies if enabled = true and only options are "source_ip_dest_ip" (2-tuple) or "source_ip_dest_ip_proto" (3-tuple).
+  # enabled = false implies 5-tuple. AWS gives type a default value of "source_ip_dest_ip_proto" even if enabled is set to false
+  stickiness {
+    enabled = var.flow_stickiness == "5-tuple" ? false : true
+    type    = var.flow_stickiness == "2-tuple" ? "source_ip_dest_ip" : "source_ip_dest_ip_proto"
+  }
 }
```

Comment:
@anichari 

GWLB (ASG and regular deployments)

- target_failover: rebalance ENABLED for both deregistration and unhealthy by default: ADD << I am prompting the user to toggle this
- stickiness: I left default as enabled=false (ie: 5-tuple) but am presenting the user the option to set/change this: ADD

Reply:
Question @jmolnar-zscaler  here on stickiness.type implementation. Looks like you have var.flow_stickiness with options 5-tuple and 2-tuple which determines enabling stickiness or not. Once enabled by selecting 2-tuple, shouldnt there be another var to ask whether to use source_ip_dest_ip or source_ip_dest_ip_proto?

Reply:
hey @anichari sorry i missed this comment. So I went the east/lazy route since the type condition only has the 2 options you mentioned.

stickiness {
    enabled = var.flow_stickiness == "5-tuple" ? false : true  <<< if user sets "5-tuple" this gets set to false. When false, the type attribute is effectively ignored.
    type    = var.flow_stickiness == "2-tuple" ? "source_ip_dest_ip" : "source_ip_dest_ip_proto"  << since this is only read when enabled = true then I can ignore 5-tuple. So my conditional statement is simply "if flow_stickiness == 2-tuple then set type to "source_ip_dest_ip", otherwise the only other option is 3-tuple ("source_ip_dest_ip_proto")
  }

Reply:
How would the user specify the vars to enable source_ip_dest_ip_proto @jmolnar-zscaler

Reply:
if var.flow_stickiness is 5-tuple then type is source_ip_dest_ip_proto(which is ignored), otherwise it is source_ip_dest_ip. There is no-way for the user to use 3-tuple

Reply:
In CFT land, I allowed all three options and based on what the user specified, I either enable/disabled and then set the type

Reply:
variable flow_stickiness has 3 constraints (that I made up): "5-tuple", "3-tuple", or "2-tuple"

option 1: user sets "5-tuple"; stickiness block gets set to:
- enabled = false
- type = "source_ip_dest_ip_proto"  << because anything not "2-tuple" falls to this. This is what AWS defaults to even if enabled is false. it does not become null.

option 2: user sets "3-tuple"; stickiness block gets set to:
- enabled = true
- type = "source_ip_dest_ip_proto"  << because anything not "2-tuple" falls to this.

option 3: user sets "2-tuple"; stickiness block gets set to:
- enabled = true
- type = "source_ip_dest_ip"  << because it matches the "2-tuple" condition

Reply:
@anichari 

type = var.flow_stickiness == "2-tuple" ? "source_ip_dest_ip" : "source_ip_dest_ip_proto"

in terraform, I can conditional expression where (var.flow_stickiness == "2-tuple") if flow_stickiness variable equals "2-tuple" then set "source_ip_dest_ip". Otherwise if flow_stickiness equals anything NOT "2-tuple" then set "source_ip_dest_ip_proto"

Reply:
got it, that works, sorry. I was confused because I didnt see a 3-tuple option.